packages/kit/types/index.d.ts

```diff
@@ -105,6 +106,15 @@ declare module '@sveltejs/kit' {
 		getServerDirectory(): string;
 		/** Get the application path including any configured `base` path, e.g. `my-base-path/_app`. */
 		getAppPath(): string;
+		/** Get the asset paths imported by server-side code. */
+		getServerAssets(): {
+			/** A map of route IDs and its array of assets paths imported by server-side code. */
+			routes: Map<string, string[]>;
+			/** An array of asset paths imported by the root [default error page](https://kit.svelte.dev/docs/routing#error). */
+			rootErrorPage: string[];
+			/** An array of asset paths imported by the [server hooks](https://kit.svelte.dev/docs/hooks#server-hooks). */
+			hooks: string[];
+		};
```

Comment:
Why is this split up into these three things? When would I want to include the assets of server hooks but not the one of the root error page? Why is the root error page even included, in the docs it says only assets from `.server` files are included?
Could it instead be just a map of routes to assets (i.e. the first property) and that one also included the assets for the hooks? Since we're using sets in the Vercel adapter this shouldn't be a problem - or did you imagine other uses cases I'm overlooking?

Reply:
> When would I want to include the assets of server hooks but not the one of the root error page? Why is the root error page even included, in the docs it says only assets from .server files are included?

The root error page uses the root layout server load, so that could include some server assets (although unlikely).

> Could it instead be just a map of routes to assets (i.e. the first property) and that one also included the assets for the hooks? Since we're using sets in the Vercel adapter this shouldn't be a problem

This is a good suggestion. I'll change it to this.

Reply:
Oh actually, we need the root error page separately so we can generate the fallback function for the vercel adapter. That function is generated by passing an empty array of routes to the generator. Only realised this after tests started failing.